modules/redis/redis.c

```diff
@@ -221,6 +213,44 @@ redis_dd_disconnect(LogThreadedDestDriver *s)
   self->c = NULL;
 }
 
+static void
+_fill_argv_from_template_list(RedisDriver *self, LogMessage *msg)
+{
+  ScratchBuffersMarker marker;
+  GString *buffer = scratch_buffers_alloc_and_mark(&marker);
+
+  for (gint i = 1; i < self->argc; i++)
+    {
+      log_template_format(g_list_nth_data(self->fields, i-1), msg, &self->template_options, LTZ_SEND,
+                          self->super.worker.instance.seq_num, NULL, buffer);
+      self->argvlen[i] = buffer->len;
+      self->argv[i] = g_strdup(buffer->str);
```

Comment:
hmm, this just invalidates the scratch buffer usage for me. does these values need to outlive the current message processing ?

Reply:
Thanks, I think I fixed it! :)